electronics_lib/PriceGetter.py

```diff
@@ -25,47 +26,58 @@ def run(self) -> Dict[str, int]:
 
 class GeneratePrice(BaseBackend):
     # part_number -> [ [lower-bound_1, price_1], [lower-bound_2 (and the previous upperbound), price_2], ... ]
-    PRICE_TABLE = Dict[str, List[Tuple[int, float]]] = {}
+    PRICE_TABLE: Dict[str, List[Tuple[int, float]]] = {}
 
     @classmethod
     def get_price_table(cls) -> Dict[str, List[Tuple[int, float]]]:
         if not GeneratePrice.PRICE_TABLE:
-            cur_path = os.path.dirname(__file__)
-            parts_library = os.path.relpath('resources/Pruned_JLCPCB SMT Parts Library(20220419).csv', cur_path)
+            current_path = os.getcwd()
+            parts_library = current_path + "\PolymorphicBlocks\electronics_lib\\resources\Pruned_JLCPCB SMT Parts Library(20220419).csv"
             with open(parts_library, 'r', newline='') as csv_file:
-                csv_reader = csv.DictReader(csv_file, 'r')
+                csv_reader = csv.reader(csv_file, dialect='excel')
+                next(csv_reader)
                 for row in csv_reader:
-                    full_price_list = row['Price']
-                    price_and_quantity_groups = full_price_list.split(',')
-                    value: List[Tuple[int, float]] = []
-                    for price_and_quantity in price_and_quantity_groups:
-                        price_and_quantity_list = price_and_quantity.split(':')
-                        assert len(price_and_quantity_list) == 2
-                        quantity_range = price_and_quantity_list[0].split('-')
-                        assert len(quantity_range) == 1 or len(quantity_range) == 2
-                        value.append((int(quantity_range[0]), float(price_and_quantity_list[1])))
+                    full_price_list = row[10]
+                    if full_price_list == " ":
+                        print("Part " + row[0] + " is missing from the price list.")
```

Comment:
Hm, how about just skipping the row (not adding a value to the table)? So that if the part isn't used, we don't see anything, and if the part is used, we can see the errors (missing from price list) with the other errors?

A robust way to parse this might be `if full_price_list.strip():` - it's a non-whitespace-sensitive test for emptyness